### PR TITLE
Cache JWKs

### PR DESCRIPTION
**Pull request by sahil-lakhwani**
- Cache JWKs

- Addresses https://github.com/infracloudio/msbotbuilder-go/issues/8

### CODE REVIEW

File: core/bot_framework_adapter.go [hunk 1]
```diff
@@ -46,7 +47,10 @@ func NewBotAdapter(settings AdapterSetting) Adapter {
 	if settings.ChannelService == "" {
 		settings.ChannelService = auth.ChannelService
 	}
-	return &BotFrameworkAdapter{settings}
+
+	jwtValidator := auth.NewJwtTokenValidator()
+
+	return &BotFrameworkAdapter{settings, jwtValidator}
```
Comment (PrasadG193) [inline core/bot_framework_adapter.go]: No need to create new variable. Could just return `&BotFrameworkAdapter{settings, auth.NewJwtTokenValidator()}

File: connector/auth/jwt_token_validation.go [hunk 2]
```diff
@@ -60,26 +60,34 @@ func (jv JwtTokenValidator) AuthenticateRequest(ctx context.Context, activity sc
...
-		return nil, err
-	}
+func (jv *JwtTokenValidator) getIdentity(authHeader string) (ClaimsIdentity, error) {
 
 	getKey := func(token *jwt.Token) (interface{}, error) {
 
-		set, err := jwk.FetchHTTP(jwksURL)
-		if err != nil {
-			return nil, err
+		if jv.jwkCache.IsExpired() {
```
Comment (PrasadG193) [inline connector/auth/jwt_token_validation.go]: Please add one-liner comments about what we are doing here

Comment (sahil-lakhwani) [reply connector/auth/jwt_token_validation.go]: Added a comment.

**Comment by sahil-lakhwani** [comment]
Closing this in favor of https://github.com/infracloudio/msbotbuilder-go/pull/24

**Pull request closed without merging**